test/e2e/util_test.go

```diff
@@ -504,11 +504,25 @@ func TearDown(namespace string) {
 	logf("test resources deleted")
 }
 
-func buildCatalogSourceCleanupFunc(crc versioned.Interface, namespace string, catalogSource *operatorsv1alpha1.CatalogSource) cleanupFunc {
+func buildCatalogSourceCleanupFunc(c operatorclient.ClientInterface, crc versioned.Interface, namespace string, catalogSource *operatorsv1alpha1.CatalogSource) cleanupFunc {
 	return func() {
 		ctx.Ctx().Logf("Deleting catalog source %s...", catalogSource.GetName())
 		err := crc.OperatorsV1alpha1().CatalogSources(namespace).Delete(context.Background(), catalogSource.GetName(), metav1.DeleteOptions{})
 		Expect(err).ToNot(HaveOccurred())
+
+		if c != nil {
+			err = wait.Poll(pollInterval, pollDuration, func() (bool, error) {
```

Comment:
The e2e suite is a bit of a weird mixture of ginkgo/gomega and testify style assertions. Recently, we've been moving to the former and utilizing the `Eventually(...)` closure to handle polling. The equivalent to wait.Poll here would be something like the following:

```go
Eventually(func() (bool, error) {
    fetched, err := ...
    if err != nil {
        return false, err
    }
    if len(fetched.Items) == 0 {
        return true, nil
    }
    return false, nil
}
```

Any thoughts?

edit: I thought more about this, and refactoring this to use the Eventually-style function shouldn't improve too much, so feel free to disregard.

Reply:
I leave it for now.